src/object/pushbutton.cpp

```diff
@@ -62,15 +63,18 @@ HitResponse
 PushButton::collision(GameObject& other, const CollisionHit& hit)
 {
   auto player = dynamic_cast<Player*>(&other);
-  if (!player) return FORCE_MOVE;
+  auto rock = dynamic_cast<Rock*>(&other);
+  if (!player && !rock) return FORCE_MOVE;
+  if (player || rock) {
```

Comment:
I don't think you need this `if` clause, since this is the only option left. You can replace it with else or delete at all.